backend/pong/oauth2/views.py

```
@@ -0,0 +1,84 @@
+    認可サーバーに接続する際に、redirect_uriを設定する理由は、認可サーバーはクライアントではなくブラウザーにレスポンスを返すため、
+    ブラウザからクライアントにリダイレクトし、認可コードを取得する必要があるため。
+
+    認可エンドポイントを呼ぶケース
+    - ユーザーが新規アカウントを作成した時
+    - ユーザーが明示的にログアウトした時(ログアウト時にアクセストークンを削除する場合)
+    - リフレッシュトークンの有効期限が切れた時
+    """
+
+    query_params = {
+        "client_id": OAUTH2_CLIENT_ID,
+        "redirect_uri": PONG_API + reverse("oauth2_callback").rstrip("/"),
```

Ruff (0.8.2)

Undefined name `PONG_API`

(F821)